source/ci_source/ci_source.ts

```diff
@@ -18,6 +18,8 @@ export interface CISource {
   /** What unique id can be found for the code review platform's PR */
   readonly pullRequestID: string
 
+  readonly commitHash?: string
```

Comment:
Not all the CI implement this at the moment, each CI will have to implement its own way of get this, for now I implemented only Jenkins and Travis

Reply:
How about other providers?

Reply:
As I said in the PR description:

> My proposal to solve the problem is to ask the CI the commit hash, and if is available use that one, otherwise fallback on the old behaviour.

Reply:
I mean it is possible to implement the `commitHash` to some providers which is available.
I know it has a lot providers to inspect. But we need to ensure github, gitlab, bitbucket server and bitbucket cloud work well at least.  According to your description, it could cause a bug. Maybe we can fix some providers we support in `danger ci`. What do you think? @orta

Reply:
What bug could it cause for the sources where is not implemented? 
If is not available it will work as before, then at the moment the first one to get this is Jenkins, and only if has the git plugin.
I will personally try it there, and then implement it in other providers with other PRs. (I already have Bitrise and Travis ready)

Reply:
>This creates few problems when you push a new commit before Danger runs on the CI for the previous commit, because it will assume is running on the last one, but that is actually not true, and that means that both the status update and the commit hash in the Danger comment signature will be wrong.
I mean some bugs already exist in other providers. So if it is possible. We could fix it now. 
If not, we could fix it later when bug is reported on other providers.

Reply:
That is how danger currently begin, we can update the other platforms, but we don't need to make all the platforms in one PR.
After we merge this i will slowly add more platforms for this, but everyone can add more platform, is actually a good "first issue" that someone can take to start contributing to the project